bio_market_store/store/templates/payment.html

```diff
@@ -46,74 +48,73 @@ <h4 class="d-flex justify-content-between align-items-center mb-3">
                             <h6 class="my-0">{{ forloop.counter }}. {{ item.name }}</h6>
                         </div>
                         <span class="text-body-secondary">{{ item.total|floatformat:2 }} PLN</span>
-
-                        {% endfor %}
+                    </li>
+                    {% endfor %}
                     <li class="list-group-item d-flex justify-content-between text-end">
                         <strong class="text-end">Total (PLN)</strong>
                         <strong>{{ cart_total|floatformat:2 }} PLN</strong>
                     </li>
                 </ul>
             </div>
 
-
             <div class="col-md-7 col-lg-8">
-
-
                 <h4 class="mb-3">Billing address</h4>
                 <form class="needs-validation" method="POST">
                     {% csrf_token %}
+
+                    <!-- Saved Addresses Dropdown -->
+                    {% if addresses %}
+                    <div class="address-selector">
+                        <label for="addressSelect" class="form-label">Saved Addresses</label>
+                        <select class="form-select" id="addressSelect">
+                            <option value="">Choose a saved address...</option>
+                        </select>
+                        <small class="form-text text-muted">Select to auto-fill address details</small>
+                    </div>
+                    {% endif %}
+
                     <div class="row g-3">
                         <div class="col-sm-6">
                             <label for="firstName" class="form-label">First name</label>
                             <input type="text" class="form-control" id="firstName" name="firstName" required>
-                            <div class="invalid-feedback">
-                                Valid first name is required.
-                            </div>
                         </div>
                         <div class="col-sm-6">
                             <label for="lastName" class="form-label">Last name</label>
                             <input type="text" class="form-control" id="lastName" name="lastName" required>
-                            <div class="invalid-feedback">
-                                Valid last name is required.
-                            </div>
                         </div>
                         <div class="col-12">
-                            <label for="username" class="form-label">Phone number</label>
+                            <label for="phoneNumber" class="form-label">Phone number</label>
                             <div class="input-group has-validation">
                                 <span class="input-group-text">+48</span>
-                                <input type="text" class="form-control" id="username" name="phoneNumber" required>
```

Comment:
Why did you remove the invalid response? (This command is required for all invalid responses).